tests/unit/test_config.py

```diff
@@ -1395,6 +1397,35 @@ def test_zero_interval(
         assert not logged
 
 
+@pytest.mark.parametrize(
+    'icp, fcp_expr, expected_fcp',
+    [
+        ('2021-02-28', '+P1M+P1D', '2021-03-29'),
+        ('2019-02-28', '+P1D+P1M', '2019-04-01'),
+        ('2008-07-01', '+P1M-P1D', '2008-07-31'),
+        ('2004-07-01', '-P1D+P1M', '2004-07-30'),
+    ]
+)
+def test_chain_expr(
+    icp: str, fcp_expr: str, expected_fcp: str, tmp_flow_config: Callable,
+):
+    """Test a "chain expression" final cycle point offset works in the
+    given order."""
+    reg = 'osgiliath'
+    flow_file: Path = tmp_flow_config(reg, f"""
+        [scheduler]
+            UTC mode = True
+            allow implicit tasks = True
+        [scheduling]
+            initial cycle point = {icp}
+            final cycle point = {fcp_expr}
```

Comment:
I'm not sure if `_parse_chain_expression` is used elsewhere, but if it is, it's worth noting that this only works if the sum of the durations > 0.

Also, should
```python
        ('2000-01-01T00:00', '+PT6H-PT1S', '2000-01-01T05:59'),
```
Work?

Reply:
- I've added more tests
- I've opened an issue for the `+PT6H+PT1S` bug: #5047 
- `parse_chain_expression()` works for net < 0 durations but that does not produce a valid final cycle point, because it is before the initial cycle point (already a tested scenario)